Road/source/features/data_cleaning.py

```diff
@@ -0,0 +1,159 @@
+# import needed libraries
+import pandas as pd 
+import numpy as np 
+import re
+import string 
+
+# Local imports
+# from source.constants import PUNCT_TO_REMOVE
+import sys
+import os
+target = os.path.abspath(__file__)
+while(target.split("\\")[-1]!="Road"):
+    target = os.path.dirname(target)
+sys.path.append(target) 
+
+# For arabic diacritics
+# pip install pyarabic 
+from pyarabic.araby import strip_tashkeel, normalize_hamza
+from pyarabic.araby import is_tashkeel, is_arabicrange
+
+# For converting emojis and emot
+# pip install emoji
+import emoji
+# pip install emot
+import emot
+
+
+def remove_diacritics(text: str) -> str:
+    """
+    Remove diacritics from Arabic text and reshape it to proper Arabic script.
+
+    Args:
+        text (str): The input Arabic text.
+
+    Returns:
+        str: The Arabic text with diacritics removed and reshaped to proper Arabic script.
+    """
+    if isinstance(text, str):
+        # Normalize the text
+        text = normalize_hamza(text)
+        # Remove diacritics using regex
+        diacritic_pattern = re.compile("[ًٌٍَُِّْ]+")
+        text_without_diacritics = diacritic_pattern.sub("", text)
+        # Reshape the text to proper Arabic script
+        reshaped_text = ""
+        for i, char in enumerate(text_without_diacritics):
+            if is_arabicrange(char):
+                if is_tashkeel(char):
+                    reshaped_text += char
+                else:
+                    reshaped_text += strip_tashkeel(char)
+            else:
+                reshaped_text += char
+        return reshaped_text
+    else:
+        return text
+
+def remove_emojis_emoticons(text: str) -> str:
+    """
+    Remove emojis and emoticons from text.
+
+    Args:
+        text (str): The input text.
+
+    Returns:
+        str: The input text with emojis and emoticons removed.
+    """
+    # remove emojis
+    text = emoji.demojize(text, delimiters=("", ""))
+    
+    # remove emoticons
+    emo_obj = emot.EMOTICONS_EMO
+    for emot1 in emo_obj:
+        escaped_emot = re.escape(emot1)
+        text = re.sub(u'({})'.format(escaped_emot), "_".join(emo_obj[emot1].replace(",", "").split()), text)
```

Comment:
What these lines do? remove emoticons or replace it with the corresponding text descriptions?

Reply:
It was replacing them. I edited the function. check it out pls.